src/common/libutil/test/base256.c

```diff
@@ -0,0 +1,110 @@
+/************************************************************\
+ * Copyright 2023 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+#include <errno.h>
+#include <string.h>
+#include <locale.h>
+#include <stdint.h>
+
+#include "src/common/libtap/tap.h"
+#include "src/common/libutil/base256.h"
+#include "ccan/str/str.h"
+
+static const char invalid[] = "😀🐨😀🌳👣🚘😹🐨";
+static const char valid[] = "🇫😀🐨😀🌳👣🚘😹🐨";
+static const char invalidemoji[] = "🇫😀🐨😀🌳👣🚘ffff";
+
+struct b256_test {
+    uint64_t data;
+    const char *result;
+} b256_tests[] =  {
+    { 0,                 "🇫😀🐨😀🐨😀🐨😀🐨" },
+    { 1,                 "🇫😁🐨😀🐨😀🐨😀🐨" },
+    { 1234,              "🇫👢🐔😀🐨😀🐨😀🐨" },
+    { 12342435,          "🇫👅🍳🗨🐨😀🐨😀🐨" },
+    { 21900760568561664, "🇫😀🐨😀🌳👣🚘😹🐨" },
```

Comment:
F58 does not populate significant digits if they are zero.  Any thoughts on whether that would be a good idea here?
```
$ flux job id --to=f58 1
ƒ2
```

Reply:
I was thinking about that too! I don't see why not, but I didn't make an attempt yet...

Reply:
I was actually thinking about that for the `words` mnemonic encoding as well. That might  be nice to shorten those too, but I haven't looked into what would be involved.